src/model/stock_predictor_manager.py

```diff
@@ -0,0 +1,218 @@
+
+        data_frame = pd.DataFrame(data, columns=["date", "value"])
+        data_frame["date"] = pd.to_datetime(data_frame["date"])
+        data_frame.set_index("date", inplace=True)
+
+        return data_frame
+
+    def _get_newest_dates(self) -> pd.DataFrame:
+        """Return last 'seq_length' dates to predict future value"""
+        data: list = []
+
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_ALL_RATE_HISTORY_DESC , [self.config.seq_length])
+            data = handler().fetchall()
+            data = [{"date": date.strftime("%Y-%m-%d"), "avg": avg} for date, avg in data]
+
+        data_frame = pd.DataFrame(data, columns=["date", "value"])
+        data_frame["date"] = pd.to_datetime(data_frame["date"])
+        data_frame.set_index("date", inplace=True)
+
+        return data_frame
+
+    def _get_train_val_data(self) -> tuple[np.ndarray, np.ndarray]:
+        """Split data into train and validation dataset."""
+        data: pd.DataFrame = self._get_data()
+
+        train_size: int = int(len(data) * self.train_size_pct)
+        val_size: int = int(len(data) * (1 - self.train_size_pct))
+
+        data_scaled = self.scaler.fit_transform(data)
+
+        train_data: np.ndarray = data_scaled[:train_size]
+        val_data: np.ndarray = data_scaled[train_size : train_size + val_size]
+
+        return train_data, val_data
+
+    def _get_data_loader(self, data: np.ndarray) -> DataLoader:
+        """Create dataloader for given dataset."""
+        data_tensor: torch.Tensor = torch.tensor(data).float()  # pylint: disable=no-member
+
+        sequence: list = []
+        for i in range(len(data_tensor) - self.config.seq_length):
+            sequence.append(
+                (
+                    data_tensor[i : i + self.config.seq_length],
+                    data_tensor[i + self.config.seq_length : i + self.config.seq_length + 1],
+                )
+            )
+
+        data_loader: DataLoader = DataLoader(
+            TensorDataset(
+                torch.stack([i[0] for i in sequence]),  # pylint: disable=no-member
+                torch.stack([i[1] for i in sequence]),  # pylint: disable=no-member
+            ),
+            batch_size=self.config.batch_size,
+        )
+
+        return data_loader
+
+    def train_model(self, verbose: bool = False) -> None:
+        """Train model"""
+
+        train_data: Optional[np.ndarray] = None
+        val_data: Optional[np.ndarray] = None
+        train_data, val_data = self._get_train_val_data()
+
+        train_loader: DataLoader = self._get_data_loader(train_data)
+        val_loader: DataLoader = self._get_data_loader(val_data)
+
+        no_improvement_count: int = 0
+        best_val_loss: float = sys.float_info.max
+        loss: torch.Tensor = torch.tensor(0)  # pylint: disable=no-member
+
+        for epoch in range(self.config.epochs):
```

Comment:
self.config.epochs -> self.config.max_epochs 
może by to zmienić w konfiguracji?

Reply:
The variable name has been changed.